### ISSUE TITLE
Suggestion: warn if sourcing a file without checking if it exists.

### ISSUE THREAD

**Issue by nihilus** [koalaman/shellcheck#1068]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
source foo/bar.script

```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Warning: sourcing without check if file exists, please use: [ -e foo/bar.script ] && source foo/bar.script

**Comment by harleypig** [comment]
Wouldn't `[ -r foo/bar.script ] && source foo/bar.script` be a better way to do this?

**Comment by Pawamoy** [comment]
It might be complicated to implement since the check could be many lines above, and the script path could be in a variable, etc.

**Comment by kurahaupo** [comment]
The recommended "fix" in the suggested warning message is bad from a correctness perspective.

If the other file is installed as part of the same package (or a prerequisite package) as the main file, such a suggestion would be misleading and worse than useless. In those cases, if the file is missing, it's a fatal error, not something that should be ignored.

If this were to be implemented, the default recommendation should be to abort with a fatal error if the other file is missing. The script writer can then decide whether it's appropriate for their case.
